libbeat/autodiscover/template/config.go

```diff
@@ -43,6 +45,10 @@ type MapperSettings []*struct {
 
 // NewConfigMapper builds a template Mapper from given settings
 func NewConfigMapper(configs MapperSettings) (mapper Mapper, err error) {
+	if len(configs) == 0 {
```

Comment:
This is a valid config, for instance when people is using hints based autodiscover, it's ok to leave templates empty:

```
filebeat.autodiscover:
  providers:
    - type: kubernetes
      hints.enabled: true
```

We would need to account for that for an error like this